### PR TITLE
BFD-3083: Remove Stateful TF modules

### PR DESCRIPTION
**Pull request by billie-rose**
**JIRA Ticket:**
[BFD-3083](https://jira.cms.gov/browse/BFD-3083)

**User Story or Bug Summary:**

As a BFD engineer, I want to remove the stateful terraform module so that we can more easily and succinctly apply changes to our terraservices.

---

### What Does This PR Do?

Move stateful terraform resources into their respective terraservice(s).
This PR is part 1 of removing the stateful and stateless terraform modules and moving away from that design paradigm to better organize our terraservices. It will also require removing the tfstate for our established environments (`test`, `prod-sbx`, and `prod`) and moving the remaining, necessary resources into their respective terraservice(s) where possible.

After this PR is accepted, we will transfer the state of the appropriate terraform resources OOB.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] The data dictionary has been updated with any field mapping changes, if any were made.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by brandoncruz3 — APPROVED** [review]

**Review by keithdadkins — APPROVED** [review]
Approved assuming all related terraform-docs have been updated!

**Review by keithdadkins — APPROVED** [review]

**Review by brandoncruz3 — APPROVED** [review]

**Review by brandoncruz3 — APPROVED** [review]

File: ops/terraform/services/common/data-sources.tf
Comment (malessi) [inline ops/terraform/services/common/data-sources.tf:83]: The names for these data resources may need to be surrounded in quotes.

File: ops/terraform/modules/resources/bfd_pipeline/main.tf
Comment (malessi) [inline ops/terraform/modules/resources/bfd_pipeline/main.tf:1]: nit: I realize this is _far_ too late to be making this comment (I thought I already had, so this is all on me), but I feel that it'd make more sense for us to pull these modules into the Terraservices directly rather than have them in a separate path altogether. The Terraservices already have submodules defined under them, so this would simply be following the existing pattern for modules used by Terraservices.

_Obviously_, this can be something we follow-up on _after_ 3083 and 3084 are merged. I don't want to hold this up because removing `stateless`/`stateful` is _way_ more important than how we organize things at the moment.

**Review by malessi — CHANGES_REQUESTED** [review]
I have a _really_ late nit that I think should be addressed in follow-up work (but I wanted to comment here for posterity) and a possible small syntax error I noticed in some data resource definitions. Due to those syntax errors I'm going to request changes for this PR, but once they're fixed everything looks great to me and I'll add my approval!

Thanks so much for cleaning this up.

File: ops/terraform/modules/resources/bfd_pipeline/main.tf
Comment (malessi) [reply ops/terraform/modules/resources/bfd_pipeline/main.tf:1]: This comment applies to the modules in `modules/resources`, specifically:

- `s3_pii`
- `cw_slack_alarms_notifier`
- `bfd_pipeline`

**Review by keithdadkins — APPROVED** [review]

**Review by malessi — APPROVED** [review]
After pairing through the `terraform plan` with Billie and verifying everything looks good and there's no unexpected changes, this looks good to me.

Thanks again Billie!

**Pull request merged**
